README.md

```diff
@@ -1,25 +1,27 @@
 # VML
 
 This package provides bindings to the Intel Vector Math Library for
-arithmetic and transcendental functions. It is often substantially
-faster than using Julia's built-in functions.
+arithmetic and transcendental functions. Especially for large vectors it is often substantially faster than broadcasting Julia's built-in functions.
 
-## Using VML.jl
+## Setting up VML.jl
 
 To use VML.jl, you must have the Intel Vector Math Library installed.
-This is included in [MKL](http://software.intel.com/en-us/intel-mkl),
-which is free for non-commercial use. You must also copy/symlink the
-appropriate shared library to a directory known to the linker (e.g.
-`/usr/local/lib`) or you must modify the path to `lib` in `src/VML.jl`.
+For this you have two options. First, you can install (and build) [MKL.jl](https://github.com/JuliaComputing/MKL.jl), which will add the necessary libraries to your Julia install. This will change your Julia system image however, so if you would prefer not to do that you can get the stand-alone [MKL](http://software.intel.com/en-us/intel-mkl),
+which is free for non-commercial use. The default install location will be detected automatically (currently only `opt/intel/mkl/lib` on Unix). If you have chosen a different location or have multiple versions of MKL, you can also specify the environmental variable 
+```
+    ENV["MKL_SL"] = <.../lib>
```

Comment:
Why not use `MKL_ROOT` for this which is what you get when you run the MKL provided script (https://software.intel.com/en-us/mkl-linux-developer-guide-scripts-to-set-environment-variables)

Reply:
When I installed MKL the MKLROOT variable was not added automatically, instead I had to add the script to my `.*rc` manually.
( Apt-get install on Ubuntu, and AUR install on arch) 
On another machine I deliberately did not add it, as the PATH was already quite long.
So I decided to just check for the default path as a fallback. 
But I agree, I should add a check for MKLROOT as well.

Thank you for reviewing this.

Reply:
But you need to set the `MKL_ROOT` to use MKL in most situations (e.g. when linking to it). To me, it seems nice to use the same way to detect MKL as everyone else is using instead of introducing our own, package specific environment variable.

Reply:
Ok, I have learned quite a bit about dynamic libaries and Libdl since I wrote this, and I have amended my commit to leave out all that stuff. 
In my quest to be as general as possible I missed that it is not really necessary. 
I think is is now basically ready.